### DISCUSSION TITLE
How to set links for the attribute values of nodes?

### DISCUSSION THREAD

**Discussion by guoranjun2** [freeplane/freeplane#1980] [Help] [answered]
I’ve been trying to figure out how to set links for node attribute values. I spent hours going through the source code but still couldn’t figure it out.

**Comment by guoranjun2** [comment]
I’ve made a bit of progress. The script below successfully sets links for the attributes:

```groovy
import org.freeplane.features.link.LinkController
import org.freeplane.features.mode.Controller
import org.freeplane.plugin.script.proxy.ScriptUtils
import org.freeplane.view.swing.map.MainView

def nodeView = ((MainView) Controller.getCurrentController().getMapViewManager().getComponent(node.delegate)).getNodeView();
def attributeView = nodeView.attributeView
def table = attributeView.getAttributeTable()
def tableModel = table.getAttributeTableModel()

String url = "https://www.google.com";
def link = LinkController.createHyperlink(url);
attributeView.startEditing()
table.setValueAt(link, 0, 1);
```
but there’s still an issue:

* When the link is set, the attribute’s text turns into a link as well.

I’ll keep digging into it.

**Comment by guoranjun2** [comment]
Well, it looks like I took the long way around. Here’s the direct approach:
```
node["link"] = org.freeplane.features.link.LinkController.createHyperlink("https://google.com")
```

**Comment by macmarrum** [comment, marked as answer]
I think `URI` will work too
```groovy
node['link'] = new URI('https://google.com')
```

**Reply by macmarrum** [reply]
Here's the relevant Scripting API doc: https://docs.freeplane.org/api/org/freeplane/api/Attributes.html

**Reply by guoranjun2** [reply]
> I think `URI` will work too
> 
> ```groovy
> node['link'] = new URI('https://google.com')
> ```

thank, this is easier,
